src/cobalt.rs

```diff
@@ -120,6 +124,57 @@ pub fn build(config: &Config) -> Result<()> {
         }
     }
 
+    // load data files
+    let mut data_map : HashMap<String, Value> = HashMap::new();
+    let data_root = source.join("_data");
+    let data_files_builder = FilesBuilder::new(data_root.as_path())?;
+    let data_files = data_files_builder.build()?;
+
+    for df in data_files.files() {
+
+        if let Some(ext) = df.extension() {
+            if let Some(file_stem) = df.file_stem() {
+                let file_name = String::from(file_stem.to_str().unwrap());
+                let full_path = data_root.join(df.clone());
+                let data : Value;
+
+                if ext == OsStr::new("yml") || ext == OsStr::new("yaml")  {
+                    let reader = File::open(full_path)?;
+                    data = serde_yaml::from_reader(reader)?;
+                } else if ext == OsStr::new("json") {
+                    let reader = File::open(full_path)?;
+                    data = serde_json::from_reader(reader).unwrap();
+                } else if ext == OsStr::new("toml") {
+                    let mut reader = File::open(full_path)?;
+                    let mut text = String::new();
+                    reader.read_to_string(&mut text)?;
+                    data = toml::from_str(&text).unwrap();
+                } else {
+                    continue
+                }
+
+                // now find the nested map it is supposed to be in
+                let mut target_map = if let Some(path) = df.parent() {
+                    let mut map = &mut data_map;
+                    for part in path.to_str().unwrap().split("/") {
```

Comment:
How come you aren't using `Path.components` or `Path.iter`? for splitting the path and iterating on it?

Reply:
Because I didn't know about them. Will take a look and learn about them. Thanks for the hint!